src/js/apps/patients/sidebar/filters-sidebar_app.js

```diff
@@ -39,10 +55,49 @@ export default App.extend({
 
     this.showChildView('customFilters', customFiltersView);
   },
+  showStatesFiltersView() {
+    const statesFiltersView = new StatesFiltersView({
+      collection: this.defaultStates,
+      state: this.getState(),
+    });
+
+    this.showChildView('statesFilters', statesFiltersView);
+  },
   onClearFilters() {
-    this.getState().clear();
+    const defaultStates = this.defaultStates.map('id').reduce((states, id) => {
+      states[id] = true;
+      return states;
+    }, {});
+
+    this.getState().clear().set({ states: defaultStates });
   },
   onStop() {
     Radio.request('sidebar', 'close');
   },
+  getFiltersCount() {
+    const state = this.getState();
+
+    const customFiltersState = omit(state.toJSON(), 'states');
+    const customFiltersSize = size(omit(customFiltersState, function(value) {
+      return isNull(value);
+    }));
+
+    const statesFilterState = state.get('states');
+    const defaultStates = this.getDefaultStates().map('id').reduce((states, id) => {
+      states[id] = true;
+      return states;
+    }, {});
+
+    const statesFilterCount = !isEqual(defaultStates, statesFilterState) ? 1 : 0;
+
+    return customFiltersSize + statesFilterCount;
+  },
+  getDefaultStates() {
+    const notDoneStates = this.states.groupByDone().notDone;
+    const doneStates = this.states.groupByDone().done;
+
+    if (this.listId === 'done-last-thirty-days') return doneStates;
+
+    return notDoneStates;
+  },
```

Comment:
These two functions (`getFiltersCount()` and `getDefaultStates()`) are identical to what's in the `filters_app.js` file.

Ideally I'd like to just pass down the `filtersCount` value to this app (`filters_app.js` => `filters-sidebar_app.js`), but I was having a heard time getting it to update in the `HeaderView`, since the `filtersCount` isn't changing along with the state.

So need to figure that out yet.